src/app/landing/shared/services/repository.service.ts

```diff
@@ -0,0 +1,72 @@
+import { Injectable, OnDestroy } from "@angular/core";
+import { CoursesByTechnology, CoursesRepository } from "../interfaces/course";
+import { mockCourses } from "./mockCourses";
+import { Observable, Subject, delay, of, retry, shareReplay, takeUntil } from "rxjs";
+import { mockStutendsFeedback } from "./mockStudentsFeedback";
+import { StudentFeedback, StudentFeedbacksRepository } from "../interfaces/student-feedback";
+import { Mentor, MentorsRepository } from "../interfaces/mentor";
+import { mockMentors } from "./mockMentors";
+
+@Injectable({providedIn: 'root'})
+export class RepositoryService implements 
+    CoursesRepository, 
+    StudentFeedbacksRepository, 
+    MentorsRepository,
+    OnDestroy
+{
+    private coursesCache: Observable<CoursesByTechnology> | null = null
+    private studentsFeedbackCache: Observable<StudentFeedback[]> | null = null
+    private mentorsCache: Observable<Mentor[]> | null = null
+
+    private unsubscribe$ = new Subject()
+
+    private prepareAPIRequest<T>(request: Observable<T>): Observable<T>{
+        return request.pipe(
+            retry({delay: 1000}),
+            takeUntil(this.unsubscribe$),
+            shareReplay(1)
+        )
+    }
+
+    private getOrFetch<T>(cache: Observable<T> | null, request: Observable<T>): Observable<T>{
+        return cache ? cache : this.prepareAPIRequest(request)
+    }
+
+    getCourses(): Observable<CoursesByTechnology>{
+        this.coursesCache = this.getOrFetch(this.coursesCache, this._getCourses())
+        return this.coursesCache
+    }
+
+    getStudentFeedbacks(): Observable<StudentFeedback[]> {
+        this.studentsFeedbackCache = this.getOrFetch(this.studentsFeedbackCache, this._getStudentFeedbacks())
+        return this.studentsFeedbackCache
+    }
+
+    getMentors(): Observable<Mentor[]> {
+        this.mentorsCache = this.getOrFetch(this.mentorsCache, this._getMentors())
+        return this.mentorsCache
```

Comment:
Check if you really need this cache logic

Reply:
I do know that **getCourses()** is called two times in a row. But i don't know how serves request will look like. Will it be one json object or three different requests. Therefore i have a trouble imagining how this service should look like.

Reply:
>  But i don't know how serves request will look like.

It should be important. You always can combine and modify API calls. Try work by layers: 
- component layer: view element and its basic logic. It gets all the necessary data from the component service layer.
- component service layer: local component state, getting/parsing/modifying data from API call layer, helpers, etc
- (component) API call layer: a service for API calls (for now it should return just hardcoded mock data)

For example, in case you should do 3 requests (3 API calls) you can combine them (for ex `combaineLatest` of `forkJoin`) in the component service:
```
combaineLatest([
  this.someRepositoryService.getDataOne(),
  this.someRepositoryService.getDataTwo(),
  this.someRepositoryService.getDataThree()
]).pipe(
  // ...
)
```